lib/int3.h

```diff
@@ -158,11 +158,11 @@ class int3
 	//returns "(x y z)" string
 	std::string toString() const
 	{
-		std::string result("(");
-		result += boost::lexical_cast<std::string>(x); result += ' ';
-		result += boost::lexical_cast<std::string>(y); result += ' ';
-		result += boost::lexical_cast<std::string>(z); result += ')';
-		return result;
+		//Performance is important here
+		char str[16] = {};
+		sprintf(str, "(%d %d %d)", x, y, z);

```

Comment:
if all coordinates have 4 digits, then you'll have no space for the null terminator.

```suggestion
		char str[17] = {};
		sprintf(str, "(%d %d %d)", x, y, z);
```

Reply:
z coordinate does have exactly one digit - it's 0 or 1 at most.

Reply:
didn't know, thanks!

Reply:
I also realistically think that map sizes above 252 x 252 are pointless, certainly no need for 4 digits.